MeshGeoToolsLib/MeshNodeSearcher.cpp

```diff
@@ -185,14 +180,11 @@ MeshNodesAlongPolyline& MeshNodeSearcher::getMeshNodesAlongPolyline(
 MeshNodesAlongSurface& MeshNodeSearcher::getMeshNodesAlongSurface(
     GeoLib::Surface const& sfc) const
 {
-    std::vector<MeshNodesAlongSurface*>::const_iterator it(
-        _mesh_nodes_along_surfaces.begin());
-    for (; it != _mesh_nodes_along_surfaces.end(); ++it)
+    for (auto const& mesh_nodes_along_surface : _mesh_nodes_along_surfaces)
     {
-        if (&(*it)->getSurface() == &sfc)
+        if (&(mesh_nodes_along_surface->getSurface()) == &sfc)
```

Comment:
Just a note. I'd think with operator== overloading for surfaces, polylines, and points, the three "equal" search functions would be equal.